src/gui/editors/PianoRoll.cpp

```diff
@@ -1587,452 +1586,488 @@ int PianoRoll::keyAreaBottom() const
-			m_pattern->splitNotes(n, TimePos(m_knifeTickPos));
-		}
 
-		update();
-		return;
-	}
-
-	if( m_editMode == ModeEditDetuning && noteUnderMouse() )
+PianoRoll::PianoRollArea PianoRoll::getPianoRollAreaIn(const int x, const int y)
+{
+	/* piano editor outline
+	 keys     | notes
+	 ---------|---------------  <-- note edit area resize bar
+	 editmode | noteproperties
+	*/
+	if (y < keyAreaBottom())
 	{
-		static QPointer<AutomationPattern> detuningPattern = nullptr;
-		if (detuningPattern.data() != nullptr)
-		{
-			detuningPattern->disconnect(this);
-		}
-		Note* n = noteUnderMouse();
-		if (n->detuning() == nullptr)
-		{
-			n->createDetuning();
-		}
-		detuningPattern = n->detuning()->automationPattern();
-		connect(detuningPattern.data(), SIGNAL(dataChanged()), this, SLOT(update()));
-		gui->automationEditor()->open(detuningPattern);
-		return;
+		if (x < m_whiteKeyWidth) { return PianoRollArea::Keys; }
+		else { return PianoRollArea::Notes; }
 	}
-
-	// if holding control, go to selection mode unless shift is also pressed
-	if( me->modifiers() & Qt::ControlModifier && m_editMode != ModeSelect )
+	else if (y <= noteEditTop())
 	{
-		m_ctrlMode = m_editMode;
-		m_editMode = ModeSelect;
-		setCursor( Qt::ArrowCursor );
-		update();
+		return PianoRollArea::EditAreaResize;
 	}
-
-	// keep track of the point where the user clicked down
-	if( me->button() == Qt::LeftButton )
+	else // > noteEditTop()
 	{
-		m_moveStartX = me->x();
-		m_moveStartY = me->y();
+		if (x < m_whiteKeyWidth) { return PianoRollArea::EditMode; }
+		else { return PianoRollArea::NoteProperties; }
 	}
+}
+
 
-	if(me->button() == Qt::LeftButton &&
-		me->y() > keyAreaBottom() && me->y() < noteEditTop())
-	{
-		// resizing the note edit area
-		m_action = ActionResizeNoteEditArea;
-		return;
-	}
 
-	if( me->y() > PR_TOP_MARGIN )
-	{
-		bool edit_note = ( me->y() > noteEditTop() );
 
-		int key_num = getKey( me->y() );
+void PianoRoll::mousePressEvent(QMouseEvent * me)
+{
+	if (!hasValidPattern()) { return; }
 
-		int x = me->x();
+	// if user clicked in the timeline, ignore
+	if (me->y() < keyAreaTop()) { return; }
 
+	const bool shiftDown = me->modifiers() & Qt::ShiftModifier;
+	// const bool ctrlDown = me->modifiers() & Qt::ControlModifier;
 
-		if (x > m_whiteKeyWidth)
-		{
-			// set, move or resize note
+	m_startedWithShift = shiftDown;
 
-			x -= m_whiteKeyWidth;
+	const bool leftButton = me->button() == Qt::LeftButton;
+	const bool rightButton = me->button() == Qt::RightButton;
 
-			// get tick in which the user clicked
-			int pos_ticks = x * TimePos::ticksPerBar() / m_ppb +
-							m_currentPosition;
+	const int mey = me->y();
+	const int mex = me->x();
 
+	// ticks per pixel = ticks per bar / pixels per bar
+	// const int tpp = static_cast<int>(static_cast<float>(TimePos::ticksPerBar()) / m_ppb);
+	// pixels per tick = pixels per bar / ticks per bar
+	const int ppt = static_cast<int>(static_cast<float>(m_ppb) / TimePos::ticksPerBar());
```

Comment:
This will truncate the decimal part and affect ticks to position calculation.

Reply:
For some reason that just didn't click in my head. 😂 I'll change to float.